src/key_packages/codec.rs

```diff
@@ -16,14 +16,16 @@ impl Codec for KeyPackage {
         let credential = Credential::decode(cursor)?;
         let extensions = extensions_vec_from_cursor(cursor)?;
         let signature = Signature::decode(cursor)?;
-        let kp = KeyPackage {
+        let mut kp = KeyPackage {
             protocol_version,
             ciphersuite: Config::ciphersuite(cipher_suite_name)?,
             hpke_init_key,
             credential,
             extensions,
             signature,
+            encoded: Vec::new(),
         };
+        kp.encoded = kp.unsigned_payload()?;
```

Comment:
It seems a bit odd to decode a key package and then immediately encode it again. But I assume this is a quirk our decoding mechanism?

Reply:
I didn't want to touch the `Cursor` to copy the encoded payload... There will be more versions of this in future that make all this nicer...

Reply:
I took the liberty of creating issue #393. Looks good to me otherwise!